src/mon/OSDMonitor.cc

```diff
@@ -687,9 +687,22 @@ OSDMonitor::update_pending_pgs(const OSDMap::Incremental& inc)
     pending_creatings.last_scan_epoch = osdmap.get_epoch();
   }
 
+  // process new pg_temp
+  auto total = pending_creatings.pgs.size();
+  // assuming the number of primed pgs cannot overload the cluster
+  for (auto& pg_temp : inc.new_pg_temp) {
+    pending_creatings.pgs.emplace(pg_temp.first,
+				  make_pair(inc.epoch,
+					    inc.modified));
+  }
```

Comment:
This seems like overkill.  pg_temp updates are common and pg creates are rare.. can we check for pg_temp in the loop below instead?  Or only insert pgs here that below to the pool(s) that are undergoing pg creation?

Reply:
but from the monitor's point of view, the pool containing this primed pg has been created. probably instead of sending pg-create, we should just update the new primary OSD with an updated osdmap?

but i am not sure what the difference is under this circumstances between these two approaches.

Reply:
also, in https://tracker.ceph.com/issues/22233, osd.3 has reported "created(pg.92.4)" message to mon. so i assume that pool 92 was added to `creating_pgs.created_pools`. hence it won't show up in `pending_creatings.pgs`.